llvm/lib/Analysis/ValueTracking.cpp

```diff
@@ -552,7 +552,9 @@ bool llvm::isValidAssumeForContext(const Instruction *Inv,
 // example Pred=EQ, RHS=isKnownNonZero. cmpExcludesZero is called in loops
 // so the extra compile time may not be worth it, but possibly a second API
 // should be created for use outside of loops.
-static bool cmpExcludesZero(CmpInst::Predicate Pred, const Value *RHS) {
+static bool cmpExcludesZero(CmpInst::Predicate Pred, Value *RHS,
+                            const SimplifyQuery *Q = nullptr,
```

Comment:
```suggestion
                            const SimplifyQuery &Q,
```
It is strange to me.  I have checked all callers of `cmpExcludeZero` and found that `SimplifyQuery` is always available.

Reply:
Its intentional so we can still have the const-only version. the perf cost of doing recursive CR call in the assume loop for example would be significant.